images/manageiq-ui-worker/container-assets/phantomjs_centos_7.sh

```diff
@@ -0,0 +1,38 @@
+# ----------------------------------------------------------------------------
+#
+# Package	: Phantomjs
+# Version	: 2.1.1
+# Source repo	: https://github.com/ariya/phantomjs/
+# Tested on	: centos_7
+# Script License: Apache License, Version 2 or later
+# Maintainer	: Priya Seth <sethp@us.ibm.com>
+#
+# Disclaimer: This script has been tested in non-root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+
+#!/bin/bash
+
+# Install all dependencies.
+sudo yum -y update
+sudo yum -y install gcc gcc-c++ make flex bison gperftools-libs \
+    ruby openssl-devel freetype-devel fontconfig-devel libicu-devel \
+    sqlite-devel libpng-devel libjpeg-devel wget git tar gzip libwebp-devel
+
+# Clone and build missing dependencies from source.
+wrkdir=`pwd`
+wget http://ftp.gnu.org/pub/gnu/gperf/gperf-3.0.4.tar.gz
+tar -xzf gperf-3.0.4.tar.gz
+cd $wrkdir/gperf-3.0.4 && ./configure && make && sudo make install
+
+# Clone PhantomJS code and build it.
+cd $wrkdir
+git clone git://github.com/ariya/phantomjs.git
```

Comment:
I'm not really a javascript person ... what is phantomjs? @himdel can you help me out here? Is this something we want to be using?

Reply:
Ah, we don't need phantomjs at runtime at all and never needed it.

Until https://github.com/ManageIQ/manageiq-ui-classic/pull/5437 (in ivanchuk+), phantomjs was needed to run `rake spec:javascript` in ui-classic,
and it is still needed to run `yarn test` in ui-service.

But if you don't need to run tests, phantomjs should not be needed :).

Reply:
>  what is phantomjs?

It's the predecessor to chrome-headless. Basically a js engine + a virtual dom implementation, to simulate a proper browser, with instrumentation.

Reply:
So, it seems to be coming from https://github.com/ManageIQ/manageiq-ui-service/blob/663dd18b3ae4fda4e09056f4281ce3be3818bae1/package.json#L128 a
nd https://github.com/ManageIQ/manageiq-ui-service/blob/663dd18b3ae4fda4e09056f4281ce3be3818bae1/package.json#L138, 
and is an attempt to install is made when we build the service ui from the manageiq-ui-worker Dockerfile, any thoughts on how to handle this?

Reply:
As above - https://github.com/ManageIQ/manageiq-pods/pull/357#discussion_r365205107 and below, I believe there are 3 choices:

* provide a fake phantomjs binary that just knows how to react to `--version`
* build SUI elsewhere, outside the container, and only copy the `yarn build` output to the target container
* install phantomjs

Reply:
(should you go with #4, `sed -i '/phantomjs/d' package.json` does the trick for me, there are no hidden dependency chains, only `phantomjs-prebuilt` and `karma-phantomjs-launcher`)